This is a testharness.js-based test.
Found 26 FAIL, 0 TIMEOUT, 0 NOTRUN.
[FAIL] createDocument test: null,";foo",null,"INVALID_CHARACTER_ERR"
  assert_throws_dom: function "function() { document.implementation.createDocument(namespace, qualifiedName, doctype) }" did not throw
[FAIL] createDocument test: null,"f}oo",null,"INVALID_CHARACTER_ERR"
  assert_throws_dom: function "function() { document.implementation.createDocument(namespace, qualifiedName, doctype) }" did not throw
[FAIL] createDocument test: null,"foo}",null,"INVALID_CHARACTER_ERR"
  assert_throws_dom: function "function() { document.implementation.createDocument(namespace, qualifiedName, doctype) }" did not throw
[FAIL] createDocument test: null,"\\ufffffoo",null,"INVALID_CHARACTER_ERR"
  assert_throws_dom: function "function() { document.implementation.createDocument(namespace, qualifiedName, doctype) }" did not throw
[FAIL] createDocument test: null,"f\\uffffoo",null,"INVALID_CHARACTER_ERR"
  assert_throws_dom: function "function() { document.implementation.createDocument(namespace, qualifiedName, doctype) }" did not throw
[FAIL] createDocument test: null,"foo\\uffff",null,"INVALID_CHARACTER_ERR"
  assert_throws_dom: function "function() { document.implementation.createDocument(namespace, qualifiedName, doctype) }" did not throw
[FAIL] createDocument test: null,"f<oo",null,"INVALID_CHARACTER_ERR"
  assert_throws_dom: function "function() { document.implementation.createDocument(namespace, qualifiedName, doctype) }" did not throw
[FAIL] createDocument test: null,":foo",null,"INVALID_CHARACTER_ERR"
  assert_throws_dom: function "function() { document.implementation.createDocument(namespace, qualifiedName, doctype) }" did not throw
[FAIL] createDocument test: null,"f:o:o",null,"INVALID_CHARACTER_ERR"
  assert_throws_dom: function "function() { document.implementation.createDocument(namespace, qualifiedName, doctype) }" threw object "NamespaceError: Failed to execute 'createDocument' on 'DOMImplementation': The namespace URI provided ('') is not valid for the qualified name provided ('f:o:o')." that is not a DOMException INVALID_CHARACTER_ERR: property "code" is equal to 14, expected 5
[FAIL] createDocument test: "",":foo",null,"INVALID_CHARACTER_ERR"
  assert_throws_dom: function "function() { document.implementation.createDocument(namespace, qualifiedName, doctype) }" did not throw
[FAIL] createDocument test: undefined,":foo",null,"INVALID_CHARACTER_ERR"
  assert_throws_dom: function "function() { document.implementation.createDocument(namespace, qualifiedName, doctype) }" did not throw
[FAIL] createDocument test: undefined,"f::oo",null,"INVALID_CHARACTER_ERR"
  assert_throws_dom: function "function() { document.implementation.createDocument(namespace, qualifiedName, doctype) }" threw object "NamespaceError: Failed to execute 'createDocument' on 'DOMImplementation': The namespace URI provided ('') is not valid for the qualified name provided ('f::oo')." that is not a DOMException INVALID_CHARACTER_ERR: property "code" is equal to 14, expected 5
[FAIL] createDocument test: "http://example.com/","fo<o",null,"INVALID_CHARACTER_ERR"
  assert_throws_dom: function "function() { document.implementation.createDocument(namespace, qualifiedName, doctype) }" did not throw
[FAIL] createDocument test: "http://example.com/",":foo",null,"INVALID_CHARACTER_ERR"
  assert_throws_dom: function "function() { document.implementation.createDocument(namespace, qualifiedName, doctype) }" did not throw
[FAIL] createDocument test: "http://example.com/","f:o:o",null,"INVALID_CHARACTER_ERR"
  assert_throws_dom: function "function() { document.implementation.createDocument(namespace, qualifiedName, doctype) }" did not throw
[FAIL] createDocument test: "http://example.com/","f::oo",null,"INVALID_CHARACTER_ERR"
  assert_throws_dom: function "function() { document.implementation.createDocument(namespace, qualifiedName, doctype) }" did not throw
[FAIL] createDocument test: "http://example.com/","0:a",null,"INVALID_CHARACTER_ERR"
  assert_throws_dom: function "function() { document.implementation.createDocument(namespace, qualifiedName, doctype) }" did not throw
[FAIL] createDocument test: "http://example.com/","a:;",null,"INVALID_CHARACTER_ERR"
  assert_throws_dom: function "function() { document.implementation.createDocument(namespace, qualifiedName, doctype) }" did not throw
[FAIL] createDocument test: "http://example.com/","a:̀",null,"INVALID_CHARACTER_ERR"
  assert_throws_dom: function "function() { document.implementation.createDocument(namespace, qualifiedName, doctype) }" did not throw
[FAIL] createDocument test: "http://example.com/","̀:a",null,"INVALID_CHARACTER_ERR"
  assert_throws_dom: function "function() { document.implementation.createDocument(namespace, qualifiedName, doctype) }" did not throw
[FAIL] createDocument test: "http://example.com/",";:a",null,"INVALID_CHARACTER_ERR"
  assert_throws_dom: function "function() { document.implementation.createDocument(namespace, qualifiedName, doctype) }" did not throw
[FAIL] createDocument test: "http://example.com/","prefix::local",null,"INVALID_CHARACTER_ERR"
  assert_throws_dom: function "function() { document.implementation.createDocument(namespace, qualifiedName, doctype) }" did not throw
[FAIL] createDocument test: "/",":foo",null,"INVALID_CHARACTER_ERR"
  assert_throws_dom: function "function() { document.implementation.createDocument(namespace, qualifiedName, doctype) }" did not throw
[FAIL] createDocument test: "http://www.w3.org/XML/1998/namespace",":foo",null,"INVALID_CHARACTER_ERR"
  assert_throws_dom: function "function() { document.implementation.createDocument(namespace, qualifiedName, doctype) }" did not throw
[FAIL] createDocument test: "http://www.w3.org/2000/xmlns/",":foo",null,"INVALID_CHARACTER_ERR"
  assert_throws_dom: function "function() { document.implementation.createDocument(namespace, qualifiedName, doctype) }" threw object "NamespaceError: Failed to execute 'createDocument' on 'DOMImplementation': The namespace URI provided ('http://www.w3.org/2000/xmlns/') is not valid for the qualified name provided (':foo')." that is not a DOMException INVALID_CHARACTER_ERR: property "code" is equal to 14, expected 5
[FAIL] createDocument test: "foo:",":foo",null,"INVALID_CHARACTER_ERR"
  assert_throws_dom: function "function() { document.implementation.createDocument(namespace, qualifiedName, doctype) }" did not throw
Harness: the test ran to completion.

